default.nix

```diff
@@ -30,63 +30,71 @@ let
     nix-store --init
   '';
 
-  build = pkgs.callPackage ./package.nix {
-    inherit nixpkgsLibPath initNix runtimeExprPath testNixpkgsPath;
-  };
 
-in
-build // {
+  results = {
+    build = pkgs.callPackage ./package.nix {
+      inherit nixpkgsLibPath initNix runtimeExprPath testNixpkgsPath;
+    };
 
-  inherit build;
+    shell = pkgs.mkShell {
+      env.NIX_CHECK_BY_NAME_EXPR_PATH = toString runtimeExprPath;
+      env.NIX_PATH = "test-nixpkgs=${toString testNixpkgsPath}:test-nixpkgs/lib=${toString nixpkgsLibPath}";
+      env.RUST_SRC_PATH = "${pkgs.rustPlatform.rustLibSrc}";
+      inputsFrom = [ results.build ];
+      nativeBuildInputs = with pkgs; [
+        npins
+        rust-analyzer
+      ];
+    };
 
-  # Used by CI and good for debugging
-  inherit pkgs;
+    # Run regularly by CI and turned into a PR
+    autoPrUpdate = pkgs.writeShellApplication {
+      name = "auto-pr-update";
+      runtimeInputs = with pkgs; [
+        npins
+        cargo
+      ];
+      text =
+        let
+          commands = {
+            "npins changes" = ''
+              npins update --directory "$REPO_ROOT/npins"'';
+            "cargo changes" = ''
+              cargo update --manifest-path "$REPO_ROOT/Cargo.toml"'';
+          };
+        in
+        ''
+          REPO_ROOT=$1
+          echo "Run automated updates"
+        ''
+        + pkgs.lib.concatStrings (pkgs.lib.mapAttrsToList (title: command: ''
+          echo -e '<details><summary>${title}</summary>\n\n```'
+          ${command} 2>&1
+          echo -e '```\n</details>'
+        '') commands);
+    };
 
-  shell = pkgs.mkShell {
-    env.NIX_CHECK_BY_NAME_EXPR_PATH = toString runtimeExprPath;
-    env.NIX_PATH = "test-nixpkgs=${toString testNixpkgsPath}:test-nixpkgs/lib=${toString nixpkgsLibPath}";
-    env.RUST_SRC_PATH = "${pkgs.rustPlatform.rustLibSrc}";
-    inputsFrom = [ build ];
-    nativeBuildInputs = with pkgs; [
-      npins
-      rust-analyzer
-    ];
+    # Tests the tool on the pinned Nixpkgs tree, this is a good sanity check
+    nixpkgsCheck = pkgs.runCommand "test-nixpkgs-check-by-name" {
```

Comment:
We're going to need an escape hatch for this.

Reply:
How do you mean?

Reply:
I mean, as we add things that cause rightfully-detected errors in nixpkgs, we'll need to allow those errors to pass CI.

Reply:
This sanity check doesn't run any ratchet checks, because it sets base Nixpkgs = main Nixpkgs :)